src/pybind/mgr/dashboard/rest_client.py

```diff
@@ -465,6 +476,12 @@ def do_request(self,
             logger.exception("%s REST API failed %s: %s", self.client_name,
                              method.upper(), str(ex))
             raise RequestException(str(ex))
+        except Timeout as ex:
+            msg = "{} REST API {} timed out after {} seconds (url={}).".format(
+                self.client_name, ex.request.method, Settings.REST_REQUESTS_TIMEOUT,
+                ex.request.url)
+            logger.exception(msg)
+            raise RequestException(msg)
```

Comment:
Too bad, we cannot use exception chaining here. https://stackoverflow.com/a/792163/3185053